### PR TITLE
Separated intersection analyses, aggregation and filter

### LINKED ISSUE #8867: Separate intersection in two separate analysis
Because depending on params it will output different kind of geometries, by separating them we'll make sure that a created analysis always returns the same kind of analysis.

Once done there should be two analysis types:
- Update option `Intersection`
  - Update desc: _"Aggregate data in a second layer as a new column."_
  - Remove measure type radio buttons
- Add option `Filter points in polygon`
  - Update desc: _Filter rows based intersection or _
  - Add form model, same as intersection but w/o any filter section (filter param should be fixed)

Considering there will be no way to change between the two once one i created, I think it's best to have separate form models and let them develop separately. Shared UI things could/should be extracted and made reusable.

### PR DESCRIPTION
**Pull request by xavijam**
screen shot 2016-07-18 at 12 24 23
screen shot 2016-07-18 at 12 24 33
screen shot 2016-07-18 at 12 24 14

Basically duplicated form model ("kind of") and only using the proper needs for each one.

Fixes #8867.
CR: @javierarce 
cc @saleiva 

- Texts have to be reviewed as @saleiva mentioned in the ticket.
- Testing analyses in staging.

### CODE REVIEW

**Comment by javierarce** [comment]
👍

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/aggregate-intersection-form-model.js [hunk 1]
```diff
@@ -87,20 +72,12 @@ module.exports = BaseAnalysisFormModel.extend({
       target: {
         type: 'NodeDataset',
         title: _t('editor.layers.analysis-form.target'),
-        options: this._getSourceOptionsForSource('target', 'point'),
+        options: this._getSourceOptionsForSource('target'),
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/aggregate-intersection-form-model.js]: Little big detail, we will accept any type of dataset in the target dropdown (aggregate intersection).
cc @saleiva

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/filter-intersection-form-model.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  /**
+   * @override {BaseAnalysisFormModel._setSchema}
+   */
+  _setSchema: function () {
+    BaseAnalysisFormModel.prototype._setSchema.call(this, this._filterSchemaFieldsByType({
+      source: this._primarySourceSchemaItem(),
+      target: {
+        type: 'NodeDataset',
+        title: _t('editor.layers.analysis-form.target'),
+        options: this._getSourceOptionsForSource('target', 'polygon'),
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/filter-intersection-form-model.js]: In the mamufas "point in polygon" analysis we will only accept polygon datasets as target.
cc @saleiva

**Comment by xavijam** [comment]
@javierarce, last 3 commits are kind of important, not really super complex things but I wanted to mention.

**Comment by xavijam** [comment]
Merging, frontend assets passed!

**Pull request merged**